R/ml_decision_tree.R

```diff
@@ -57,25 +79,48 @@ ml_decision_tree <- function(x,
   schema <- sdf_schema(df)
   responseType <- schema[[response]]$type
 
-  regressor  <- "org.apache.spark.ml.regression.DecisionTreeRegressor"
-  classifier <- "org.apache.spark.ml.classification.DecisionTreeClassifier"
-
-  envir$model <- if (identical(type, "regression"))
-    regressor
-  else if (identical(type, "classification"))
-    classifier
-  else if (responseType %in% c("DoubleType", "IntegerType"))
-    regressor
-  else
-    classifier
-
-  rf <- invoke_new(sc, envir$model)
-
-  model <- rf %>%
+  modelType <- if (identical(type, "regression"))
```

Comment:
Do you think it's worth moving this code (+ the code around `impurity`) to a set of helper functions?